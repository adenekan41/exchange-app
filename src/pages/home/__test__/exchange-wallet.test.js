/* eslint-disable no-undef */
/* -------------------------- External Dependencies ------------------------- */
import React from 'react';
import ReactDOM from 'react-dom';
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';

import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Router } from 'react-router-dom';

/* -------------------------- Internal Dependencies ------------------------- */
import history from 'utils/history';
import ExchangeWallet from '../components/exchange-wallet';

const mockStore = configureStore([]);

describe('ExchangeWallet Component', () => {
	afterEach(cleanup);
	let store;
	beforeAll(() => {
		store = mockStore({
			exchange: {
				currentRate: 100,
				from: 'USD',
				to: 'EUR',
				wallet: {
					USD: {
						amount: 400,
					},
					GBP: {
						amount: 500,
					},
					EUR: {
						amount: 700,
					},
				},
			},
		});
		store.dispatch = jest.fn();
	});
	it('ExchangeWallet Renders Without Crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(
			<Provider store={store}>
				<Router history={history}>
					<ExchangeWallet />
				</Router>
			</Provider>,
			div
		);
		ReactDOM.unmountComponentAtNode(div);
	});
	it('ExchangeWallet matches snapshot', () => {
		const tree = renderer
			.create(
				<Provider store={store}>
					<Router history={history}>
						<ExchangeWallet />
					</Router>
				</Provider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('ExchangeWallet matches snapshot without actions', () => {
		const tree = renderer
			.create(
				<Provider store={store}>
					<Router history={history}>
						<ExchangeWallet />
					</Router>
				</Provider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
