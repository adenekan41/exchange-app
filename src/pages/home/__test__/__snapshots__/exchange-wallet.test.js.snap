// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangeWallet Component ExchangeWallet matches snapshot 1`] = `
<section
  className="sc-AxiKw dGZWJM"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-lg"
      >
        <div
          className="sc-AxhCb bjIare card"
        >
          <div
            className="card-body"
          >
            <div
              className="sc-AxirZ dLMRmj"
              data-testid="select"
            >
              
              <select
                className="form-control"
                id="input"
                onChange={[Function]}
                value="USD"
              >
                <option
                  disabled={false}
                  value="USD"
                >
                  $
                   
                  USD
                </option>
                <option
                  disabled={true}
                  value="EUR"
                >
                  €
                   
                  EUR
                </option>
                <option
                  disabled={false}
                  value="GBP"
                >
                  £
                   
                  GBP
                </option>
              </select>
              <svg>
                arrow-down.svg
              </svg>
            </div>
            <p
              className="pills"
            >
              Wallet Balance
            </p>
            <h1>
              $
               
              400.00
            </h1>
          </div>
          <div
            className="conversion__card"
          >
            <div
              className="sc-AxjAm boKlPC"
              data-testid="input"
            >
              <label
                htmlFor="Exchange USD -> EUR"
              >
                Exchange USD -&gt; EUR
              </label>
              <input
                className="form-control"
                data-testid="inputs"
                id="0"
                name="from"
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="0.00"
                type="number"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-lg-2 d-md-flex align-items-center my-5 my-md-0"
      >
        <button
          className="btn btn-primary btn-block"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Exchange
        </button>
      </div>
      <div
        className="col-lg"
      >
        <div
          className="sc-AxhCb bjIare card"
        >
          <div
            className="card-body"
          >
            <div
              className="sc-AxirZ dLMRmj"
              data-testid="select"
            >
              
              <select
                className="form-control"
                id="input"
                onChange={[Function]}
                value="EUR"
              >
                <option
                  disabled={true}
                  value="USD"
                >
                  $
                   
                  USD
                </option>
                <option
                  disabled={false}
                  value="EUR"
                >
                  €
                   
                  EUR
                </option>
                <option
                  disabled={false}
                  value="GBP"
                >
                  £
                   
                  GBP
                </option>
              </select>
              <svg>
                arrow-down.svg
              </svg>
            </div>
            <p
              className="pills"
            >
              Wallet Balance
            </p>
            <h1>
              €
               
              700.00
            </h1>
          </div>
          <div
            className="conversion__card"
          >
            <div
              className="sc-AxjAm boKlPC"
              data-testid="input"
            >
              <label
                htmlFor="Exchange USD -> EUR"
              >
                Exchange USD -&gt; EUR
              </label>
              <input
                className="form-control"
                data-testid="inputs"
                id="0"
                name="from"
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="0.00"
                type="number"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`ExchangeWallet Component ExchangeWallet matches snapshot without actions 1`] = `
<section
  className="sc-AxiKw dGZWJM"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-lg"
      >
        <div
          className="sc-AxhCb bjIare card"
        >
          <div
            className="card-body"
          >
            <div
              className="sc-AxirZ dLMRmj"
              data-testid="select"
            >
              
              <select
                className="form-control"
                id="input"
                onChange={[Function]}
                value="USD"
              >
                <option
                  disabled={false}
                  value="USD"
                >
                  $
                   
                  USD
                </option>
                <option
                  disabled={true}
                  value="EUR"
                >
                  €
                   
                  EUR
                </option>
                <option
                  disabled={false}
                  value="GBP"
                >
                  £
                   
                  GBP
                </option>
              </select>
              <svg>
                arrow-down.svg
              </svg>
            </div>
            <p
              className="pills"
            >
              Wallet Balance
            </p>
            <h1>
              $
               
              400.00
            </h1>
          </div>
          <div
            className="conversion__card"
          >
            <div
              className="sc-AxjAm boKlPC"
              data-testid="input"
            >
              <label
                htmlFor="Exchange USD -> EUR"
              >
                Exchange USD -&gt; EUR
              </label>
              <input
                className="form-control"
                data-testid="inputs"
                id="0"
                name="from"
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="0.00"
                type="number"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-lg-2 d-md-flex align-items-center my-5 my-md-0"
      >
        <button
          className="btn btn-primary btn-block"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Exchange
        </button>
      </div>
      <div
        className="col-lg"
      >
        <div
          className="sc-AxhCb bjIare card"
        >
          <div
            className="card-body"
          >
            <div
              className="sc-AxirZ dLMRmj"
              data-testid="select"
            >
              
              <select
                className="form-control"
                id="input"
                onChange={[Function]}
                value="EUR"
              >
                <option
                  disabled={true}
                  value="USD"
                >
                  $
                   
                  USD
                </option>
                <option
                  disabled={false}
                  value="EUR"
                >
                  €
                   
                  EUR
                </option>
                <option
                  disabled={false}
                  value="GBP"
                >
                  £
                   
                  GBP
                </option>
              </select>
              <svg>
                arrow-down.svg
              </svg>
            </div>
            <p
              className="pills"
            >
              Wallet Balance
            </p>
            <h1>
              €
               
              700.00
            </h1>
          </div>
          <div
            className="conversion__card"
          >
            <div
              className="sc-AxjAm boKlPC"
              data-testid="input"
            >
              <label
                htmlFor="Exchange USD -> EUR"
              >
                Exchange USD -&gt; EUR
              </label>
              <input
                className="form-control"
                data-testid="inputs"
                id="0"
                name="from"
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="0.00"
                type="number"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
